src/blueapi/service/config_manager.py

```diff
@@ -0,0 +1,26 @@
+# config_manager.py
+
+from blueapi.config import ApplicationConfig
+
+
+class ConfigManager:
+    """Manages application configuration in a way that’s easy to test and mock."""
+
+    _config: ApplicationConfig
+
+    def __init__(self, config: ApplicationConfig = None):
+        if config is None:
+            ApplicationConfig.model_config["yaml_file"] = None
```

Comment:
Should: Again, I think you should find an alternative to mutating global state, if you fix the instance in CLI then you shouldn't need this line.